students/C_Farris/session05/Mailroom_3.py

```diff
@@ -0,0 +1,211 @@
+    :Param: none
+    returns: donation retrieved from user
+    """
+    getDonation = ''
+    while not type(getDonation) is int:
+        try:
+            getDonation = int(input("Please enter donation amount:"
+                                    "==>"))
+        except ValueError:
+                print('Sorry, that isn\'t a valid integer. Please retry')
+    return getDonation
+
+
+def thankyou():
+    """
+    :param:
+    :returns:
+    """
+    getPerson = ''
+    donation = ''
+    while getPerson not in donor_db:
+        getPerson = input('Please type the donors first and last name or '
+                          'type list to get donor list ==>')
+        getPerson = getPerson.strip().lower().title()
+        if getPerson == 'List':
+            for key in donor_db:
+                print(key)
+        elif getPerson == 'x':
+            exitout()
+        else:
+            try:
+                donation = getDonation()
+                donor_db[getPerson].append(donation)
+            except KeyError:
+                print("This new person will be added to the dictionary.")
+                donation = getDonation()
+                donor_db[getPerson] = [donation]
+            finally:
+                printThankYou(getPerson, donation)
+
+
+def alldonors():
+    """
+    Print thank you letters to all donors.
+    first retrieve donor list, print to last donation and collective donations.
+    will have to pass the person and the donation amount to "print thank you"
+    """
+    print("###############################################################")
+    print("Printing to file thank you letters to all donors in database...")
+    print("###############################################################")
+    for key, value in donor_db.items():
+        sumDonations = round(float(sum(value)), 2)  # total Given
+        donorLetter = assemble_thank_you(key, sumDonations)
+        send_file_to_disk(key, donorLetter)
+
+
+def printThankYou(getPerson, getDonation):
+    """
+    prints thank you letter for person and donation specified
+    :param: User provided person and donation amount
+    :returns: none (will return letter to save to disk shortly)
+    """
+    print(assemble_thank_you(getPerson, getDonation))
+
+
+def makereport():
+    """
+    Calculates values to put into report
+    :param:
+    :return:
+    """
+    donorReport = [[round(float(sum(value)), 2), key, len(value),
+                    round(float(sum(value) / len(value)), 2)]
+                    for key, value in donor_db.items()]
+    sortedReport = sorted(donorReport)
+    ascendingReport = sortedReport[::-1]
+    printReport(ascendingReport)
+
+
+def printReport(ascendingReport):
+    """
+    print report using f.strings
+    :param (name): sorted list in ascending order that
+                   contains required information
+                   for each donor
+    :return: It prints, neet to put into a return statement####
```

Comment:
yup!

I like to put comments in like:

```
# fixme: need to put into a return statement
```
then I can search for "fixme" to see stuff I need to do.